crates/yttrium/src/uniffi_compat/log.rs

```diff
@@ -0,0 +1,48 @@
+use {std::sync::Arc, tracing_subscriber::fmt::MakeWriter};
+
+#[cfg(feature = "uniffi")]
+#[derive(Clone)]
+struct UniffiLogger(Arc<dyn Logger>);
+
+#[cfg(feature = "uniffi")]
+impl<'a> MakeWriter<'a> for UniffiLogger {
+    type Writer = UniffiLogger;
+
+    fn make_writer(&'a self) -> Self::Writer {
+        self.clone()
+    }
+}
+
+#[cfg(feature = "uniffi")]
+impl std::io::Write for UniffiLogger {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let message = String::from_utf8_lossy(buf);
+        self.0.log(message.to_string());
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+#[cfg(feature = "uniffi")]
+#[uniffi::export(with_foreign)]
+pub trait Logger: Send + Sync {
+    fn log(&self, message: String);
+}
+
+#[cfg(feature = "uniffi")]
```

Comment:
Don't need these fine-grained conditionals because this entire module `uniffi_compat` only is enabled for the `uniffi` feature